### PR TITLE
Fix PyQt5 compatibility

### PR DESCRIPTION
**Pull request by pankajp**
Move PyQt5 out of allowed failures on travis-ci and fix a few issues

### CODE REVIEW

**Comment by corranwebster** [comment]
Do you feel comfortable that the Qt5 tests won't segfault now?

**Comment by pankajp** [comment]
Oh, is that failure intermittent? I fixed a specific segfault, and it passed twice in a row so we can try our luck and the next time pyqt5 segfaults we move it to allowed failures again.

**Comment by corranwebster** [comment]
Yes, failure was intermittent, but if you've fixed a particular segfault it's probably worth a shot at seeing if it is stable.

File: traitsui/qt4/table_editor.py [hunk 1]
```diff
@@ -1251,3 +1265,11 @@ def _update_filter_list(self):
         filters = self.filters
         self.filters = []
         self.filters = filters
+
+if is_qt5:
```
Comment (corranwebster) [inline traitsui/qt4/table_editor.py]: Minor style suggestion: put this at the top of the file (possibly immediately after the definition of `is_qt5`).

**Pull request merged**